src/v2/BaseLoginRouter.js

```diff
@@ -103,7 +103,16 @@ export default Router.extend({
     //    -> refresh page
     //    -> introspect using options.stateHandle
     if (lastResponse) {
-      sessionStorageHelper.setStateHandle(idxResponse?.context?.stateHandle);
+      const stateHandle = idxResponse?.context?.stateHandle;
+      const newAppId = idxResponse?.context?.app?.value?.id;
+      const oldAppId = sessionStorageHelper.getAppId();
+
+      if (oldAppId && (oldAppId !== newAppId)) {
+        sessionStorageHelper.removeStateHandle();
+      }
```

Comment:
Can you add use cases (in PR description or diagram to `docs` folder) ?
I'm kind of curious how session isn't clean previously

Reply:
I think this is happening due to user copy pasting a new url into the existing and intends that new url should login correctly. In this case the stored token is still valid, but user's intention is to actually login to a different app.. So we have to discard the existing stored stateToken if AppID is different..

Reply:
It was not a bug. Just a missed use case.

Reply:
@haishengwu-okta PR description contains a demo video of this change and a sentence describing the new handled case.